### DISCUSSION TITLE
Is possible to include these two things in the user.js?

### DISCUSSION THREAD

**Discussion by ghost** [yokoffing/Betterfox#227] [General]
Going through  the 'optional hardening' section found this:

Sanitize on close

Clear browsing data on shutdown:

    Go to ☰ → Settings → Privacy & Security → Cookies and Site Data → Delete cookies and site data when Firefox is closed

You can allow exceptions if you want to stay logged in to some sites:

    Go to ☰ → Settings → Privacy & Security → Cookies and Site Data → Manage Exceptions
    For cross-domain logins, add exceptions for both sites
        e.g. https://www.youtube.com (site) + https://accounts.google.com (single sign on)

Wondering if these two things, "Delete cookies and site data when Firefox is closed" and the list of exception domains can be done via user.js file instead of manually.

**Comment by yokoffing** [comment]
> Delete cookies and site data when Firefox is closed

It used to be `user_pref("network.cookie.lifetimePolicy", 2);`, but that was removed.

Currently, toggling on that setting in the UI flips the following prefs to `true`:
```
privacy.sanitize.sanitizeOnShutdown
privacy.clearOnShutdown.cache
privacy.clearOnShutdown.cookies
privacy.clearOnShutdown.offlineApps
```

But if you want everything cleared (minus site exceptions if you use them), then you'll need:
```
// PREF: clear browsing data on shutdown
user_pref("privacy.sanitize.sanitizeOnShutdown", true);
user_pref("privacy.clearOnShutdown.history", true);
user_pref("privacy.clearOnShutdown.formdata", true);
user_pref("privacy.clearOnShutdown.sessions", true);
user_pref("privacy.clearOnShutdown.cache", true);
user_pref("privacy.clearOnShutdown.cookies", true);
user_pref("privacy.clearOnShutdown.offlineApps", true);
user_pref("privacy.clearOnShutdown.siteSettings", false); // enforce
user_pref("browser.sessionstore.privacy_level", 2);
```

If you have session restore enable (_Settings → General → Startup → Opened previous windows and tabs_), then you will need to omit `privacy.clearOnShutdown.history` for session restore to work.

> list of exception domains can be done via user.js

https://github.com/yokoffing/Betterfox/blob/0334bea9721fa734506160f0770515817c094972/Securefox.js#L467-L470

**Reply by ghost** [reply]
Thanks man, that works perfectly, I definitely will go through all the files to learn more about Firefox and the fantastic work you are doing here. Totally a fan.

**Reply by ghost** [reply]
About this: 

 For cross-domain logins, add exceptions for both sites:
// e.g. https://www.youtube.com (site) + https://accounts.google.com (single sign on)

I just logged on youtube and just allow as exception accounts.google, after that and clearing all except site.Setting and history, youtube and all google related webs login automatically just with accounts.google as exception (no need to add the site too)

**Reply by yokoffing** [reply]
> login automatically just with `accounts.google` as exception (no need to add the site too)

I have observed this behavior myself for a long time, but I've kept it in the documentation for now. It doesn't hurt either way, but there may be some nuanced difference I've overlooked.
